etstool.py

```diff
@@ -309,7 +309,15 @@ def flake8(edm, runtime, toolkit, environment):
 
     """
     parameters = get_parameters(edm, runtime, toolkit, environment)
-    commands = ["{edm} run -e {environment} -- python -m flake8"]
+    targets = [
```

Comment:
Instead of this, I'd suggest changing the flake8 settings in a config file. That way the settings apply regardless of how `flake8` is run, and not just in CI.

(I presume that this change is because there's some file that needs to be excluded from the `flake8` run?)

Reply:
It actually wasn't a single file with errors, gh actions was trying to run flake8 on everything in the edm environment and was running for 20+ min:
```
./.edm/pkgs/9c/5dfc3b11de9f6cf06b47cbe5f431d4f56bebe5fe9c1c14492052301c13cd51/pip/_vendor/chardet/langcyrillicmodel.py:264:12: E231 missing whitespace after ','
./.edm/pkgs/9c/5dfc3b11de9f6cf06b47cbe5f431d4f56bebe5fe9c1c14492052301c13cd51/pip/_vendor/chardet/langcyrillicmodel.py:264:14: E231 missing whitespace after ','
./.edm/pkgs/9c/5dfc3b11de9f6cf06b47cbe5f431d4f56bebe5fe9c1c14492052301c13cd51/pip/_vendor/chardet/langcyrillicmodel.py:264:16: E231 missing whitespace after ','
./.edm/pkgs/9c/5dfc3b11de9f6cf06b47cbe5f431d4f56bebe5fe9c1c14492052301c13cd51/pip/_vendor/chardet/langcyrillicmodel.py:264:18: E231 missing whitesp
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```

Reply:
Actually, I don't think I understand why this change was made - `flake8` seems to run fine for me on the entire codebase; could you comment further?

Reply:
Hmm. That's weird. How did `.edm` end up in the current directory?

Reply:
Ah, I see; this is related to setting the EDM root directory. But putting the EDM root inside the repository clone doesn't seem right here. Is there something else we could do?

How are other ETS packages managing to avoid this issue?

Reply:
We recently made the switch to Github actions in pyface and apptools.  Pyface doesn't flake8 on CI, but apptools is what actually inspired the change I made here. 
TraitsUI does style checks without using edm at all:
https://github.com/enthought/traitsui/blob/master/.github/workflows/style-checks.yml

Reply:
Also the code:
```
env:
      # Set root directory, mainly for Windows, so that the EDM Python
      # environment lives in the same drive as the cloned source. Otherwise
      # 'pip install' raises an error while trying to compute
      # relative path between the site-packages and the source directory.
      EDM_ROOT_DIRECTORY: ${{ github.workspace }}/.edm
```
has been copied around _pretty much_ every ets project that has started using GH actions.  Admittedly I personally have done so blindly, assuming it was needed for some reason and not investigating more

Reply:
> copied around _pretty much_ every ets project that has started using GH actions

Well, not Traits or Traits Futures, which were the first ETS projects to use GH actions.  :-)

We should see if we can find a way to avoid this. The need to modify the way we run `flake8` shows that this arrangement has undesirable side-effects; other tools may similarly be affected (e.g., if we wanted to run `black`, or `mypy`, or `isort`).

One option may simply be to use the `path` attribute in `actions/checkout`, so that the repository ends up being cloned into its own directory instead  of the top level of the workspace. That's then closer to how we work with the code as developers.

Do you have an example of the `pip` error that appears from the relative path issue anywhere? I'm wondering whether there are other workarounds possible. This seems like something that ought to be a known issue for `pip` (and if not, we may want to open an issue on the `pip` tracker).

Reply:
> Do you have an example of the pip error that appears from the relative path issue anywhere?

I do not currently, but I will do some digging

Reply:
It was included in the first commit for the relevant file in traitsUI:
https://github.com/enthought/traitsui/pull/1492/commits/4823b3f0e76bb3f43a3309a7471e042f0940e8af
and I'm not seeing any references to the error. I opened a quick PR without it on traitsUI to see if I see an error:
https://github.com/enthought/traitsui/pull/1697

Reply:
Thanks. I don't want to derail this PR - we can merge this one as-is, but it would  still be good to figure out  a better solution sometime.

Reply:
SGTM

Thank you for pointing this out! Far better than it just going unacknowledged

Reply:
> Perhaps it is not needed there? and maybe not here either

Ah, interesting. Could we try removing it here to see what happens? (Can do that in a separate PR if you prefer.)

Also, I note from the enable traceback that the pip install seems to be using pip's "legacy" install mode; that might be because `wheel` isn't installed, and could explain why this issue doesn't seem to show up in the pip bug tracker.

Reply:
I went ahead and removed it here.  I had already turned off ets-from-source and test-wx workflows on PR runs, but I figure test-with-edm is enough to verify if it is needed or not.